ngraph/core/src/op/experimental_detectron_roi_feature.cpp

```diff
@@ -0,0 +1,136 @@
+    visitor.on_attribute("sampling_ratio", m_attrs.sampling_ratio);
+    visitor.on_attribute("pyramid_scales", m_attrs.pyramid_scales);
+    visitor.on_attribute("aligned", m_attrs.aligned);
+    return true;
+}
+
+void op::v6::ExperimentalDetectronROIFeatureExtractor::validate_and_infer_types()
+{
+    NGRAPH_OP_SCOPE(v6_ExperimentalDetectronROIFeatureExtractor_validate_and_infer_types);
+    NODE_VALIDATION_CHECK(this, get_input_size() >= 2, "At least two argument required.");
+
+    auto rois_shape = get_input_partial_shape(0);
+    auto input_et = get_input_element_type(0);
+
+    PartialShape out_shape = {
+        Dimension::dynamic(), Dimension::dynamic(), m_attrs.output_size, m_attrs.output_size};
+
+    PartialShape out_rois_shape = {Dimension::dynamic(), 4};
+    if (rois_shape.rank().is_static())
+    {
+        NODE_VALIDATION_CHECK(
+            this, rois_shape.rank().get_length() == 2, "Input rois rank must be equal to 2.");
+
+        NODE_VALIDATION_CHECK(this,
+                              rois_shape[1].is_static() && rois_shape[1].get_length() == 4u,
+                              "The last dimension of the 'input_rois' input must be equal to 4. "
+                              "Got: ",
+                              rois_shape[1]);
+
+        out_shape[0] = rois_shape[0];
+        out_rois_shape[0] = rois_shape[0];
+    }
+
+    size_t num_of_inputs = get_input_size();
+    std::vector<Dimension> channels(num_of_inputs - 1);
+
+    for (size_t i = 1; i < num_of_inputs; i++)
+    {
+        auto& channel = channels[i - 1];
+        auto current_shape = get_input_partial_shape(i);
+        auto current_rank = current_shape.rank();
+
+        if (current_rank.is_dynamic() || current_shape[1].is_dynamic())
+        {
+            set_output_type(0, input_et, out_shape);
+            return;
+        }
+
+        NODE_VALIDATION_CHECK(this,
+                              current_rank.get_length() == 4,
+                              "Rank of each element of the pyramid must be equal to 4. Got: ",
+                              current_rank);
+
+        NODE_VALIDATION_CHECK(this,
+                              current_shape[0].is_static() && current_shape[0].get_length() == 1u,
+                              "The first dimension of each pyramid element must be equal to 1. "
+                              "Got: ",
+                              current_shape[0]);
+
+        channel = current_shape[1];
+    }
+
+    auto featmap_shape = get_input_partial_shape(1);
+    auto expected_channels = featmap_shape[1];
+
+    bool correct_channels =
+        std::all_of(channels.begin() + 1, channels.end(), [&expected_channels](const Dimension& d) {
+            return expected_channels == d;
+        });
+
+    NODE_VALIDATION_CHECK(this,
+                          correct_channels,
+                          "The number of channels must be the same for all layers of the pyramid.");
+
+    out_shape[1] = expected_channels;
```

Comment:
We always assume that the model is correct, so the logic of checking channels value could be modified the following way.
1. We find intersection of all channel dimensions (using the `&` operation). 
2. Check that this Interval is not empty (using method `empty` of the `Interval` class). If it is empty that means that the model is not correct.
3. Set the corresponding output dimension to be equal to the interval calculated in the step 1.

Reply:
Make sure that you cover this case with unit tests properly.